sdk/appconfiguration/azure-data-appconfiguration/src/configuration_client.cpp

```diff
@@ -0,0 +1,2316 @@
+    {
+      if (i != 0)
+      {
+        valueStr += "&tags=";
+      }
+
+      valueStr += Core::Url::Encode(options.Tags[i]);
+    }
+
+    url.AppendQueryParameter("tags", valueStr);
+  }
+
+  Core::Http::Request request(Core::Http::HttpMethod::Head, url);
+
+  request.SetHeader("Accept", "application/json");
+
+  request.SetHeader("Sync-Token", options.SyncToken);
+  request.SetHeader("Accept-Datetime", options.AcceptDatetime);
+  request.SetHeader("If-Match", options.IfMatch);
+  request.SetHeader("If-None-Match", options.IfNoneMatch);
+  request.SetHeader("x-ms-client-request-id", options.XMsClientRequestId.ToString());
+  auto rawResponse = m_pipeline->Send(request, context);
+  auto const httpStatusCode = rawResponse->GetStatusCode();
+
+  if (httpStatusCode != Core::Http::HttpStatusCode::Ok)
+  {
+    throw Core::RequestFailedException(rawResponse);
+  }
+
+  CheckKeyValuesResult response{};
+  response.SyncToken = rawResponse->GetHeaders().at("Sync-Token");
+  response.ETag = rawResponse->GetHeaders().at("ETag");
+
+  return Response<CheckKeyValuesResult>(std::move(response), std::move(rawResponse));
+}
+
+Azure::Response<GetKeyValueResult> ConfigurationClient::GetKeyValue(
+    std::string const& key,
+    std::string const& accept,
+    GetKeyValueOptions const& options,
+    Core::Context const& context)
+{
+  auto url = m_url;
+  url.AppendPath("kv/");
+  url.AppendPath(!key.empty() ? Core::Url::Encode(key) : "null");
+
+  url.AppendQueryParameter("api-version", Core::Url::Encode(m_apiVersion));
+  url.AppendQueryParameter("label", Core::Url::Encode(options.Label));
+  {
+    std::string valueStr;
+    for (size_t i = 0; i < options.Select.size(); ++i)
+    {
+      if (i != 0)
+      {
+        valueStr += ',';
+      }
+
+      valueStr += Core::Url::Encode(options.Select[i].ToString());
+    }
+
+    url.AppendQueryParameter("$Select", valueStr);
+  }
+
+  Core::Http::Request request(Core::Http::HttpMethod::Get, url);
+
+  request.SetHeader("Accept", accept);
+  request.SetHeader("Sync-Token", options.SyncToken);
+  request.SetHeader("Accept-Datetime", options.AcceptDatetime);
+  request.SetHeader("If-Match", options.IfMatch);
+  request.SetHeader("If-None-Match", options.IfNoneMatch);
+  request.SetHeader("x-ms-client-request-id", options.XMsClientRequestId.ToString());
+  auto rawResponse = m_pipeline->Send(request, context);
+  auto const httpStatusCode = rawResponse->GetStatusCode();
+
+  if (httpStatusCode != Core::Http::HttpStatusCode::Ok)
+  {
+    throw Core::RequestFailedException(rawResponse);
+  }
+
+  GetKeyValueResult response{};
+  {
+    auto const& responseBody = rawResponse->GetBody();
+    if (responseBody.size() > 0)
+    {
+      auto const jsonRoot
+          = Core::Json::_internal::json::parse(responseBody.begin(), responseBody.end());
+
+      response.Key = jsonRoot["key"].get<std::string>();
+
+      if (jsonRoot.contains("label") && !jsonRoot["label"].is_null())
```

Comment:
This throws a weird undiagnosable exception if the label property in the JSON object is not a string. Does it make sense to check the type so you can generate a better exception?

OTOH, this is also validating a server response, so maybe it's not a big deal.

Reply:
I do believe our stance isn't to validate the JSON schema (or other such format) of the returned response from the service, if it deviates from the spec.

If we do want a resilient JSON desrialization of responses in the future (not just valid JSON but valid schema), we'd have to adjust our approach to the generated (and hand-written) implementations.

Reply:
We could consider wrapping all `json.get<std::string>();` calls with a helper that does the following, but I am not sure if it's worth doing (I initially thought of that to avoid leaking nlohmann exceptions to the end user, but that's a larger topic):
```C++
if (!json.is_string())
  throw RFE
else
  json.get<std::string>();
```

That said, the following fails with a fairly reasonable exception:
```C++
auto const jsonRoot = Azure::Core::Json::_internal::json::parse("{\"foo\": 1}");
EXPECT_EQ(jsonRoot["foo"].get<std::string>(), "bar");
```
```text
error: C++ exception with description "[json.exception.type_error.302] type must be string, but is number" thrown in the test body.

```